src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -125,7 +126,37 @@ def create_launcher(fluent_launch_mode: LaunchMode = None, **kwargs):
         return SlurmLauncher(**kwargs)
 
 
+def deprecate(old_arg, new_arg, converter):
+    """Deprecates old argument and replaces them with corresponding new argument."""
+
+    def decorator(func):
+        """Holds the original method to perform operations on it."""
+
+        def wrapper(*args, **kwargs):
+            """Performs deprecation operation on arguments of the original method."""
+            if old_arg in kwargs:
+                converter(old_arg, new_arg, kwargs)
+                kwargs.pop(old_arg)
+            return func(*args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
+def _show_gui_to_ui_mode_converter_func(old_arg, new_arg, kwargs):
+    warnings.warn(
+        "'show_gui' is deprecated. Use 'ui_mode' instead.",
+        PyFluentDeprecationWarning,
+    )
+    if kwargs[old_arg]:
```

Comment:
A few thoughts here. 
- We could create a general base class (or function) that issues the warning as long as it is provided with the names of the arguments.
- Should we be more strict about the type of `show_gui` (in the above line)? It's already the case that it can be any type. The same is true for other arguments like `start_transcript`.

Reply:
yes @seanpearsonuk, we should do something regarding the type of "show_gui". Will update the code.

Reply:
Could you also check te same for similar arguments. Something as simple as `if x is True` for `if x` and `if x is False` for `if not x`.